### PR TITLE
WIP: serde 1.0

### PR DESCRIPTION
**Pull request by rubdos**
W.r.t. https://github.com/rust-num/num/pull/302 & #218, I took the arrow to the knee and started changing what needed changing.

~~TODO:~~ Done; that was quicker than I thought.

- [x] BigInt
- [x] Complex
- [x] Rational

Did I miss more `serde` stuff?

### CODE REVIEW

**Comment by oli-obk** [comment]
Travis still has a `cargo update -p serde --precise 0.7.0` which probably needs to be updated ;)

**Comment by rubdos** [comment]
> Travis still has a cargo update -p serde --precise 0.7.0 which probably needs to be updated ;)

Did I miss something else? :-)

**Comment by oli-obk** [comment]
Rust 1.8 isn't supported anymore. Serde needs 0.13 I think (at least that's what our travis checks)

**Comment by rubdos** [comment]
> Rust 1.8 isn't supported anymore. Serde needs 0.13 I think (at least that's what our travis checks)

So remove that too?

**Comment by rubdos** [comment]
Seems like I had forgot a file.

**Comment by rubdos** [comment]
Seem to pass now. Is there a reason though, to still support 1.13?

**Comment by cuviper** [comment]
I consider it a breaking change to bump the minimum rust, so please don't
remove 1.8 testing entirely here. Just add 1.13 to CI as the minimum for
using serde.

(Even though we *are* making a breaking change here, let's isolate that
change to serde for the moment.)

File: bigint/Cargo.toml [hunk 1]
```diff
@@ -29,7 +29,7 @@ version = "0.3.19"
 
 [dependencies.serde]
 optional = true
-version = ">= 0.7.0, < 0.9.0"
+version = ">= 1.0.0, < 2.0.0"
```
Comment (cuviper) [inline bigint/Cargo.toml]: Please make this a simple "1.0" dependency.

**Comment by cuviper** [comment]
Thanks!

**Pull request merged**